spec/unit/file_system_spec.rb

```diff
@@ -5,6 +5,14 @@
 describe "Puppet::FileSystem" do
   include PuppetSpec::Files
 
+  def with_file_content(content)
+    path = tmpfile('file-system')
+    file = File.new(path, 'wb')
+    file.sync = true
+    file.print content
+    yield path
```

Comment:
we should explicitly close the file handle so we don't rely on finalizers, or alternatively use the block form of `File.open`